tests/countries/test_nepal.py

```diff
@@ -0,0 +1,536 @@
+        ):
+            self.assertHolidayName(name, dt)
+
+    def test_prithvi_jayanti(self):
+        name = "Prithvi Jayanti"
+        self.assertHolidayName(name, (f"{year}-01-11" for year in range(2023, 2050)))
+        self.assertNoHolidayName(name, range(2010, 2023))
+
+    def test_martyrs_day(self):
+        name = "Martyr's Day"
+
+        # Public Holidays.
+        self.assertHolidayName(
+            name, (f"{year}-01-30" for year in (*range(2010, 2019), *range(2021, 2050)))
+        )
+        self.assertNoHolidayName(name, range(2019, 2021))
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-01-30" for year in range(2019, 2021))
+        )
+        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2050))
+
+    def test_national_democracy_day(self):
+        name = "National Democracy Day"
+
+        # Public Holidays.
+        self.assertHolidayName(
+            name, (f"{year}-02-19" for year in (*range(2010, 2019), *range(2021, 2050)))
+        )
+        self.assertNoHolidayName(name, range(2019, 2021))
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-02-19" for year in range(2019, 2021))
+        )
+        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2050))
+
+    def test_womens_day(self):
+        self.assertHolidayName("Women's Day", (f"{year}-03-08" for year in range(2010, 2050)))
+
+    def test_labor_day(self):
+        self.assertHolidayName("Labor Day", (f"{year}-05-01" for year in range(2010, 2050)))
+
+    def test_republic_day(self):
+        name = "Republic Day"
+
+        # Public Holidays.
+        self.assertHolidayName(
+            name, (f"{year}-05-29" for year in (*range(2010, 2019), *range(2021, 2050)))
+        )
+        self.assertNoHolidayName(name, range(2019, 2021))
+
+        # Workdays.
+        self.assertHolidayName(
+            name, self.workday_holidays, (f"{year}-05-29" for year in range(2019, 2021))
+        )
+        self.assertNoHolidayName(name, self.workday_holidays, range(2010, 2019), range(2021, 2050))
+
+    def test_constitution_day(self):
+        self.assertHolidayName("Constitution Day", (f"{year}-09-19" for year in range(2010, 2050)))
+
+    def test_christmas(self):
+        self.assertHolidayName("Christmas", (f"{year}-12-25" for year in range(2010, 2050)))
+
+    def test_tamu_losar(self):
+        self.assertHolidayName("Tamu Losar", (f"{year}-12-30" for year in range(2010, 2050)))
+
+    def test_bakrid(self):
+        name = "Bakrid"
+        dt = (
+            "2019-08-12",
+            "2020-08-01",
+            "2021-07-21",
+            "2022-07-10",
+            "2023-06-29",
+            "2024-06-17",
+            "2025-06-07",
+        )
+        self.assertHolidayName(name, dt)
+        self.assertHolidayName(name, range(2010, 2026))
```

Comment:
```suggestion
        self.assertHolidayName(name, self.no_estimated_holidays, range(2010, 2050))
```
And same for Eid al-Fitr.

It's much more practical if the tests are arranged in the same order as the holidays.